### PR TITLE
controller/podautoscaler: minor cleanup

### PR DESCRIPTION
**Pull request by xiang90**
1. remove unnecessary else statement
2. remove unnecessary initialization

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/28759)

### CODE REVIEW

**Comment by xiang90** [comment]
@bprashanth Can you please take a look?

**Comment by xiang90** [comment]
@bprashanth Can you review the change? Or can you please reassign?

**Comment by bprashanth** [comment]
Sorry for the delay, didn't realize this was assigned to me. Assigned to Jerzy who is more familiar.

File: pkg/controller/podautoscaler/horizontal.go [hunk 1]
```diff
@@ -352,22 +348,24 @@ func (a *HorizontalController) reconcileAutoscaler(hpa *autoscaling.HorizontalPo
...
-		// Going down only if the usageRatio dropped significantly below the target
-		// and there was no rescaling in the last downscaleForbiddenWindow.
-		if desiredReplicas < currentReplicas &&
-			(hpa.Status.LastScaleTime == nil ||
-				hpa.Status.LastScaleTime.Add(downscaleForbiddenWindow).Before(timestamp)) {
-			return true
-		}
+	if desiredReplicas == currentReplicas {
+		return false
+	}
```
Comment (jszczepkowski) [inline pkg/controller/podautoscaler/horizontal.go]: I would remove this if, we have return false at the end.

**Comment by jszczepkowski** [comment]
Thanks for the cleanup. LGTM

**Pull request merged**